[UIK-4029][data-table] added renderCellOverlay (#2379)
semrush/intergalactic: semcore/data-table/CHANGELOG.md

- `line-height` for `Head.Column` items.
- `overflow=hidden` for `CellWrapper`.
- `textAlign` property to all column cells.

> - `renderCellOverlay` method for render some card over table cells.

Reviewer: ```suggestion
- `renderCellOverlay` method for rendering an overlay over table cells.
```
